sdk/eventhub/event-hubs/src/receiver.ts

```diff
@@ -232,61 +262,25 @@ export class EventHubConsumer {
    */
   async receiveBatch(
     maxMessageCount: number,
-    maxWaitTimeInSeconds?: number,
+    maxWaitTimeInSeconds: number = 60,
     abortSignal?: AbortSignalLike
   ): Promise<ReceivedEventData[]> {
     this._throwIfReceiverOrConnectionClosed();
     this._throwIfAlreadyReceiving();
-    const checkpoint = this.getCheckpoint();
-    if (checkpoint) {
-      this._eventPosition = EventPosition.fromSequenceNumber(checkpoint);
-    }
-    if (!this._batchingReceiver || !this._batchingReceiver.isOpen()) {
-      this._batchingReceiver = BatchingReceiver.create(
-        this._context,
-        this.consumerGroup,
-        this.partitionId,
-        this._eventPosition,
-        this._receiverOptions
-      );
-    } else if (this._batchingReceiver.checkpoint < checkpoint!) {
-      await this._batchingReceiver.close();
-      this._batchingReceiver = BatchingReceiver.create(
-        this._context,
-        this.consumerGroup,
-        this.partitionId,
-        this._eventPosition,
-        this._receiverOptions
-      );
-    }
 
-    let result: ReceivedEventData[] = [];
     try {
-      result = await this._batchingReceiver.receive(
-        maxMessageCount,
-        maxWaitTimeInSeconds,
-        this._receiverOptions.retryOptions,
-        abortSignal
-      );
-      if (result.length < maxMessageCount) {
-        // we are now re-using the same receiver link between multiple calls to receiveBatch() or the iterator on the receiver.
-        // This can result in the receiver link having pending credits if a receiveBatch() call asking for m events returned n events where n < m.
-        // Since Event Hubs doesnt support the drain feature yet, this can result in the receiver link receiving events when the user is not expecting it to.
-        // Hence closing the link when result.length < maxMessageCount
-        await this._batchingReceiver.close();
-      }
+      return await this._receiveBatch(maxMessageCount, maxWaitTimeInSeconds, abortSignal);
     } catch (err) {
       log.error(
         "[%s] Receiver '%s', an error occurred while receiving %d messages for %d max time:\n %O",
         this._context.connectionId,
-        this._batchingReceiver.name,
+        this._baseConsumer && this._baseConsumer.name,
```

Comment:
In `_receiveBatch` the entire relevant code is wrapped in a promise which is then passed to `retry()`
`retry()` internally has a try/catch around the given promise and will catch anything that is thrown.

Therefore, this entire catch -> log -> throw in `receiveBatch()` seems unnecessary
Once removed, there isn't much happening here other than calling `_recieveBatch()`, so I would suggest removing `_receiveBatch()` and move the code inside `recieveBatch()`

Reply:
Sounds fair! I did this in order to keep the log message on errors the same as the current implementation, but I would be happy to remove it!